uint/src/uint.rs

```diff
@@ -346,7 +346,58 @@ pub fn split_u128(a: u128) -> (u64, u64) {
 
 #[macro_export]
 macro_rules! construct_uint {
+	( $(#[$attr:meta])* $visibility:vis struct $name:ident (1); ) => {
+		construct_uint!{ @construct $($attr)* $visibility struct $name (1); }
+	};
+
 	( $(#[$attr:meta])* $visibility:vis struct $name:ident ( $n_words:tt ); ) => {
+			construct_uint! { @construct $($attr)* $visibility struct $name ($n_words); }
+
+			impl $crate::core_::convert::From<u128> for $name {
+				fn from(value: u128) -> $name {
+					let mut ret = [0; $n_words];
+					ret[0] = value as u64;
+					ret[1] = (value >> 64) as u64;
+					$name(ret)
+				}
+			}
+
+			impl $crate::core_::convert::From<i128> for $name {
+				fn from(value: i128) -> $name {
+					match value >= 0 {
```

Comment:
this could be `match (n as u128) >> 127 == 0` but the assembly is identical according to https://rust.godbolt.org/z/5kXf8K, so better keep it way more readable